tests/TypeAnnotationTest.php

```diff
@@ -189,6 +189,46 @@ function toArray(): array {
                         }
                     }'
             ],
+            'classTypeAliasImportWithAlias' => [
+                '<?php
+                    /** @psalm-type PhoneType = array{phone: string} */
+                    class Phone {
+                        /** @psalm-return PhoneType */
+                        public function toArray(): array {
+                            return ["phone" => "Nokia"];
+                        }
+                    }
+
+                    /**
+                     * @psalm-import-type PhoneType from Phone as TPhone
+                     */
+                    class User {
+                        /** @psalm-return TPhone */
+                        function toArray(): array {
+                            return array_merge([], (new Phone)->toArray());
+                        }
+                    }'
+            ],
+            'classTypeAliasDirectUsage' => [
+                '<?php
+                    /** @psalm-type PhoneType = array{phone: string} */
+                    class Phone {
+                        /** @psalm-return PhoneType */
+                        public function toArray(): array {
+                            return ["phone" => "Nokia"];
+                        }
+                    }
+
+                    /**
+                     * @psalm-import-type PhoneType from Phone
```

Comment:
„Phone“ is a class-name , I guess. Is a test required with a fqcn?

Reply:
Added on https://github.com/vimeo/psalm/pull/3375/commits/24b2dbc716a5a693185d222e5cef078a0b48eb59